tests/integration/test_chat.py

```diff
@@ -912,3 +918,9 @@ def test_ping_handler_when_chat_is_complete_with_step_without_responses(self, mo
         self.click_button('OK')
         self.wait_for_ajax()
         self.assertTrue(mock_handler.called)
+        # The handler should be called again if we reload the page.
+        mock_handler.reset_mock()
+        self.assertFalse(mock_handler.called)
+        self.element = self.go_to_view("student_view")
+        self.wait_for_ajax()
+        self.assertTrue(mock_handler.called)
```

Comment:
Yes, the tests could use some work, but I'm trying to keep this task as small as possible, so I'll leave that for another time.